microsetta_interface/implementation.py

```diff
@@ -813,25 +812,89 @@ def get_create_human_source(*, account_id=None):
     if has_error:
         return consent_output
 
+    form_type = "source"
+
     return _render_with_defaults(
         'new_participant.jinja2',
         tl=consent_output,
         post_url=post_url,
         duplicate_email_check_url=duplicate_email_check_url,
         home_url=home_url,
+        form_type=form_type,
         language_tag=session_locale())
 
 
 @prerequisite([ACCT_PREREQS_MET])
 def post_create_human_source(*, account_id=None, body=None):
-    has_error, consent_output, _ = ApiRequest.post(
-        "/accounts/{0}/consent".format(account_id), json=body)
-    if has_error:
-        return consent_output
 
-    new_source_id = consent_output["source_id"]
+    create_new_source = False
+    consent_type = body.get("consent_type")
+
+    if "Data" in consent_type:
+        consent_type = "Data"
+
+        if "source_id" in session:
+            source_id = session['source_id']
+
+            dup_check_body = {}
+            dup_check_body["participant_name"] = body.get("participant_name")
+
+            has_error, email_check_output, _ = ApiRequest.post(
+                "/accounts/{0}/check_duplicate_source".format(
+                    account_id), json=dup_check_body)
+
+            if has_error:
+                return email_check_output
+
+            if email_check_output['source_duplicate']:
+                has_error, consent_required, _ = ApiRequest.get(
+                    '/accounts/%s/source/%s/consent/%s' % (account_id, source_id, 'Data'))
+
+                if consent_required["result"]:
+                    has_error, consent_output, _ = ApiRequest.post(
```

Comment:
It looks like `has_error` is not handled on these calls, should it be?

Reply:
Corrected!

Reply:
[Consent Task Status.docx](https://github.com/biocore/microsetta-interface/files/9903923/Consent.Task.Status.docx)
@wasade the doc has multiple screenshots

Reply:
Is it possible to see screenshots of what the document looks like rendered by microsetta-interface?

Reply:
it was shared